### PR TITLE
bg animation of conway's game of life  - fixes #48

### LINKED ISSUE #48: bg ascii art: conway's game of life
same as https://www.google.com/search?q=conway%27s+game+of+life

### PR DESCRIPTION
**Pull request by umutbasal**

### CODE REVIEW

**Comment by umutbasal** [comment]
- [ ] ~~debounce on handleResize~~
    - [ ] ~~h,w use effect deps~~
- [x] livingCells re renders all cells
- [ ] ~~setTimeout to requestAnimationFrame~~
- [x] ~~react-virtualized~~
- [x] canvas


canvas worked great

**Comment by umutbasal** [comment]
`https://playgameoflife.com/lexicon/data/B-heptomino.json`
https://playgameoflife.com/lexicon/B-heptomino
`"pattern": "O.OO\nOOO.\n.O..\n"`

to convert coords and place to mid
```
	function parsePatternToCoords(pattern: string, midX: number, midY: number) {
		const coords = [];
		
		const rows = pattern.trim().split('\n');
		
		const height = rows.length;
		const width = rows[0].length;
		
		const offsetX = Math.floor(width / 2);
		const offsetY = Math.floor(height / 2);
		
		for (let y = 0; y < height; y++) {
			for (let x = 0; x < width; x++) {
				if (rows[y][x] === 'O') {
					coords.push([midX + (x - offsetX), midY + (y - offsetY)]);
				}
			}
		}
		
		return coords;
	}
```

File: components/custom/game-of-life.tsx [hunk 1]
```diff
@@ -0,0 +1,194 @@
+"use client";
+import React, { useState, useEffect, useRef, useCallback } from "react";
```
Comment (ozantekin) [inline components/custom/game-of-life.tsx]: Using `* as React` format here ensures consistency across the project.

Comment (umutbasal) [reply components/custom/game-of-life.tsx]: done

**Pull request merged**